### PR TITLE
Separate Caption and Detection Models

### PR DESCRIPTION
**Pull request by yashkant**

### CODE REVIEW

File: captioning/detect_caption.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+from .utils import get_detectron_features
+from pythia.common.sample import Sample, SampleList
+import gc
+import torch
+
+# Wrapper class over Detection and Captioning model that defines the
+# combined the caption generation from raw image path
```
Comment (abhshkdz) [inline captioning/detect_caption.py:7]: can't parse this sentence

File: captioning/detect_caption.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                self.cuda_device
+            )
+            # returns a single-element list
+            detectron_features = detectron_features[0]
+
+            sample = Sample()
+            sample.dataset_name = "coco"
+            sample.dataset_type = "test"
+            sample.image_feature_0 = detectron_features
+            sample.answers = torch.zeros((5, 10), dtype=torch.long)
```
Comment (abhshkdz) [inline captioning/detect_caption.py:33]: what's this `(5, 10)` for?

Comment (abhshkdz) [reply captioning/detect_caption.py:33]: Also not clear why anything to do with `answers` has to be inside this captioning wrapper.

File: captioning/detect_caption.py (hunk 1 above)
Comment (yashkant) [reply captioning/detect_caption.py:7]: :laughing:  sorry, it reads: 

Wrapper class over Detection (feat-extraction) and Captioning model, that defines `predict` function which takes in the imagepath and generates the caption.

Comment (yashkant) [reply captioning/detect_caption.py:7]: Fixing this in the latest commit.

File: captioning/detect_caption.py (hunk 2 above)
Comment (yashkant) [reply captioning/detect_caption.py:33]: Yeah, it makes me uncomfortable too :disappointed: , this is coming from the `predict` function of the colab demo we used as reference: 
https://colab.research.google.com/drive/1vzrxDYB0vxtuUy8KCaGxm--nDCJvyBSg#scrollTo=UCD0nso8YelA&forceEdit=true&offline=true&sandboxMode=true

Comment (yashkant) [reply captioning/detect_caption.py:33]: After digging inside, I find that  `sample.answers` is used in the forward pass of captioning model on this line: 
https://github.com/facebookresearch/pythia/blob/master/pythia/models/butd.py#L113

**Pull request merged**